server/src/main/java/org/elasticsearch/ingest/PipelineExecutionService.java

```diff
@@ -78,7 +78,7 @@ protected void doRun() throws Exception {
                             innerExecute(indexRequest, getPipeline(indexRequest.getPipeline()));
                             //this shouldn't be needed here but we do it for consistency with index api
                             // which requires it to prevent double execution
-                            indexRequest.setPipeline(null);
+                            indexRequest.setPipeline("");
```

Comment:
Had to change this since `null` now triggers an empty loop because it causes the default pipeline to be set over and over.

Reply:
Can you elaborate on what triggers this? I'm unsure about the original comment, but it seems like that should be fixed first, so that we can have `null` as meaning "no pipeline" in the index request. Empty string should be an error. We need some type of literal string (which we disallow from pipeline names) to mean "no pipeline". In the original default pipelines issue, I suggested `_none` or something like that. Using empty string is too easy to happen with a malformed request (like accidentally not serializing http params correctly into the request).

Reply:
@rjernst the problem is that I need to somehow differentiate between "no pipeline so set default" (when the request first hits `org.elasticsearch.action.bulk.TransportBulkAction#doExecute`) and "no pipeline, period" (when it hits `org.elasticsearch.action.bulk.TransportBulkAction#doExecute` and I want to prevent the default to be applied again). Previously double execution here could be prevented by setting `null`, but now I used `null` to trigger setting the default and `""` to mean that no default should be set (since that is also what I'd see if someone used a url ending in `?pipeline=`).

So better move to `_none` for when we want to prevent setting a pipeline and also use the default on `""` (i.e. URLs ending in `?pipeline=`)? :)

Reply:
> So better move to _none for when we want to prevent setting a pipeline and also use the default on "" (i.e. URLs ending in ?pipeline=)? :)

Sorry I don't understand what you are suggesting here.

Reply:
I feel strongly `?pipeline=` should be an error. It is too easy too have someone mean to override a pipeline, but then get the default (which is the opposite of the problem when using `?pipeline=` to mean don't run any pipeline).

Reply:
@rjernst I like it :) Error for `?pipeline=` it is :) => on it